src/emit/solana/target.rs

```diff
@@ -2135,14 +2152,18 @@ impl<'a> TargetRuntime<'a> for SolanaTarget {
                     )
                     .into()
             }
-            codegen::Expression::Builtin(_, _, codegen::Builtin::Balance, args) => {
-                assert_eq!(args.len(), 1);
+            codegen::Expression::Builtin {
+                builtin: codegen::Builtin::Balance,
+                args: values,
```

Comment:
Why didn't you use `args`?

Reply:
I called this `values` first and then changed my mind. I've fixed it.